### PR TITLE
*: replace interface{} with any

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #4399

### What is changed and how does it work?
see https://github.com/pingcap/tidb/pull/50770

```commit-message
```

### Check List

Tests

- Unit test
only refactor

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: client/client.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ type Client interface {
 	// WatchGlobalConfig returns a stream with all global config and updates
 	WatchGlobalConfig(ctx context.Context, configPath string, revision int64) (chan []GlobalConfigItem, error)
 	// UpdateOption updates the client option.
-	UpdateOption(option DynamicOption, value interface{}) error
+	UpdateOption(option DynamicOption, value any) error
```
Comment (rleungx) [inline client/client.go:137]: How about adding some fmt checks? See https://golangci-lint.run/usage/linters/#gofmt

Comment (CabinfeverB) [reply client/client.go:137]: CI works https://github.com/tikv/pd/actions/runs/7708423098/job/21007463045?pr=7774

**Review by nolouch — APPROVED** [review]
lgtm!

File: .golangci.yml [hunk 2]
```diff
@@ -41,3 +42,10 @@ linters-settings:
       - require-error
       - suite-dont-use-pkg
       - suite-extra-assert-call
+  gofmt:
+    # https://golangci-lint.run/usage/linters/#gofmt
+    # disable for faster check
+    simplify: false
+    rewrite-rules:
+      - pattern: 'interface{}'
+        replacement: 'any'
```
Comment (rleungx) [inline .golangci.yml:51]: need an empty line

**Review by rleungx — APPROVED** [review]

**Pull request merged**